src/Resources/Base/Resource.php

```diff
@@ -123,7 +123,7 @@ public function getName()
     {
         $reflection = new ReflectionClass($this);
 
-        return $this->snakeCase($reflection->getShortName());
+        return $reflection->getShortName();
```

Comment:
@frodosghost I wasn't sure about this change. Do you remember why the `->snakeCase()` call was in there? Removing it seems to be okay, the existing endpoints function in the same way.

Reply:
From memory, it was to set the URL to be correct when it queries for the Resource.

Calling `->getShortName()` returns the capitalised class, so for the `SmartOysters\FarmOpsX\Resources\ViewReportsHarvestReports` class it shows `ViewReportsHarvestReports`. But, we didn't want that for the class name, we wanted it returned in [snake_case](https://en.wikipedia.org/wiki/Snake_case).

I think, for URL calls, this would mean that endpoint would be `view_reports_harvest_reports`. Although, that doesn't seem to be much of an issue for the other endpoints we have, as all the models map to the single, lowercase naming convention.

Reply:
@frodosghost I have been trying to get the endpoint to be underscored in the Reporting API. ChatGPT suggested adding `"plural": "view_reports_harvest_reports"` into the model, this did not change it. It also suggested adding the plural option into `model-config.json`, which also didn't work.

Reply:
@liamclaridge444 :: I wouldn't bother too much with changing Loopback - that is archaic technology.

In this situation I would probably overwrite the `getName()` function in the extended class. That way most of the classes use the base from this class, but the new one can use this adjustment, with the `return $reflection->getShortName();`